rs/nns/governance/src/neuron/types.rs

```diff
@@ -875,7 +906,7 @@ impl Neuron {
             || self.visibility() == Some(Visibility::Public)
             || self.is_hotkey_or_controller(&requester);
         if show_full {
-            recent_ballots.append(&mut self.recent_ballots.clone());
+            recent_ballots.append(&mut self.recent_ballots());
```

Comment:
This doesn't seem to take into account the ordering of recent_ballots which may be reversed depending on if you have set `self.recent_ballots_next_entry_index`.